### PR TITLE
Tidy up validate-modules ignores for some cloud/ modules

### PR DESCRIPTION
**Pull request by russoz**
##### SUMMARY
Tidy up on sanity tests' ignore files, removing lines with validation-modules for these modules.

The focus was to pass the tests so no code was changed other than adjusting argument_spec for missing types or inconsistencies with the docs. The priority has been not to change any functionality, but to ensure that code and docs are synchronized.

Some lines were added back to the files, but a total of 103 lines were removed (among 2.9, 2.10 and 2.11).

##### ISSUE TYPE
- Docs Pull Request

##### COMPONENT NAME
plugins/modules/cloud/xenserver/xenserver_guest_powerstate.py
plugins/modules/cloud/xenserver/xenserver_guest_info.py
plugins/modules/cloud/dimensiondata/dimensiondata_vlan.py
plugins/modules/cloud/dimensiondata/dimensiondata_network.py
plugins/doc_fragments/dimensiondata.py
plugins/modules/cloud/linode/linode_v4.py
plugins/modules/cloud/linode/linode.py
plugins/modules/cloud/memset/memset_zone_record.py
plugins/modules/cloud/memset/memset_zone.py
plugins/modules/cloud/memset/memset_zone_domain.py
plugins/modules/cloud/memset/memset_server_info.py
plugins/modules/cloud/memset/memset_memstore_info.py
plugins/modules/cloud/memset/memset_dns_reload.py
plugins/modules/cloud/misc/terraform.py
plugins/modules/cloud/misc/proxmox_template.py
plugins/modules/cloud/misc/proxmox_kvm.py
plugins/modules/cloud/softlayer/sl_vm.py
plugins/modules/cloud/ovh/ovh_ip_loadbalancing_backend.py
plugins/modules/cloud/ovh/ovh_ip_failover.py
plugins/modules/cloud/huawei/hwc_vpc_subnet.py
plugins/modules/cloud/huawei/hwc_vpc_port.py
plugins/modules/cloud/huawei/hwc_ecs_instance.py
plugins/modules/cloud/pubnub/pubnub_blocks.py
plugins/modules/cloud/docker/docker_stack.py
plugins/modules/cloud/heroku/heroku_collaborator.py
plugins/modules/cloud/oracle/oci_vcn.py
plugins/modules/cloud/atomic/atomic_container.py

### CODE REVIEW

File: plugins/doc_fragments/dimensiondata.py [hunk 1]
```diff
@@ -19,7 +19,6 @@ class ModuleDocFragment(object):
   region:
     description:
       - The target region.
-    choices:
```
Comment (felixfontein) [inline plugins/doc_fragments/dimensiondata.py:22]: That's the most interesting `choices` values I've ever seen. I had to check out how it looks in the docs: https://docs.ansible.com/ansible/2.10/collections/community/general/dimensiondata_network_module.html#parameter-region

File: plugins/modules/cloud/linode/linode.py [hunk 2]
```diff
@@ -97,28 +112,34 @@
...
   kernel_id:
     description:
      - kernel to use for the instance (Linode Kernel)
+    type: int
   wait:
     description:
      - wait for the instance to be in state C(running) before returning
     type: bool
-    default: "no"
+    default: true
```
Comment (felixfontein) [inline plugins/modules/cloud/linode/linode.py:137]: Yes, no, who cares... :)

**Review by felixfontein — APPROVED** [review]
LGTM

File: tests/sanity/ignore-2.11.txt [hunk 3]
```diff
@@ -42,15 +41,6 @@ plugins/modules/cloud/centurylink/clc_server_snapshot.py validate-modules:doc-mi
...
 plugins/modules/cloud/centurylink/clc_server_snapshot.py validate-modules:parameter-list-no-elements
 plugins/modules/cloud/centurylink/clc_server_snapshot.py validate-modules:parameter-type-not-in-doc
-plugins/modules/cloud/dimensiondata/dimensiondata_network.py validate-modules:doc-choices-do-not-match-spec
-plugins/modules/cloud/dimensiondata/dimensiondata_network.py validate-modules:doc-missing-type
-plugins/modules/cloud/dimensiondata/dimensiondata_network.py validate-modules:parameter-type-not-in-doc
-plugins/modules/cloud/dimensiondata/dimensiondata_vlan.py validate-modules:doc-choices-do-not-match-spec
-plugins/modules/cloud/dimensiondata/dimensiondata_vlan.py validate-modules:doc-missing-type
-plugins/modules/cloud/dimensiondata/dimensiondata_vlan.py validate-modules:doc-required-mismatch
-plugins/modules/cloud/dimensiondata/dimensiondata_vlan.py validate-modules:parameter-type-not-in-doc
-plugins/modules/cloud/docker/docker_container.py use-argspec-type-path # uses colon-separated paths, can't use type=path
```
Comment (felixfontein) [inline tests/sanity/ignore-2.11.txt:52]: This line needs to stay (with the comment). See the sanity errors.

**Review by krsacme — APPROVED** [review]

File: plugins/modules/cloud/linode/linode.py (hunk 2 above)
Comment (russoz) [reply plugins/modules/cloud/linode/linode.py:137]: ahahaha

**Comment by felixfontein** [comment]
ignore-2.9.txt: https://app.shippable.com/github/ansible-collections/community.general/runs/5387/11/tests

**Comment by russoz** [comment]
Those two parameters actually have no documentation whatsoever... look like legacy stuff people forgot to deprecate, but then again, I haven't analyzed the whole thing.

**Review by felixfontein — APPROVED** [review]

**Comment by felixfontein** [comment]
@russoz thanks for fixing these as well!
@krsacme thanks for reviewing!

**Comment by russoz** [comment]
Thanks @felixfontein and @krsacme !

File: plugins/modules/cloud/misc/proxmox_template.py [hunk 4]
```diff
@@ -49,7 +49,6 @@
     description:
       - Proxmox VE node, when you will operate with template
     type: str
-    required: true
```
Comment (Aversiste) [inline plugins/modules/cloud/misc/proxmox_template.py:52]: This change seems wrong, without the `node` parameter it is not possible to reach the necessary API calls.

Comment (felixfontein) [reply plugins/modules/cloud/misc/proxmox_template.py:52]: In that case it's not the docs that are wrong, but the code. The argument spec (implicitly) says that this is not required: https://github.com/russoz/community.general/blob/27a11715129311dc11fed1283e755e462f1c5bcc/plugins/modules/cloud/misc/proxmox_template.py#L213

Comment (Aversiste) [reply plugins/modules/cloud/misc/proxmox_template.py:52]: Ah yes, I can see that. One more thing on my TODO list then.

**Pull request merged**